addons/frag/functions/fnc_addPfhRound.sqf

```diff
@@ -1,30 +1,32 @@
 //#define DEBUG_MODE_FULL
 #include "script_component.hpp"
 
-private ["_enabled", "_doSpall", "_spallTrack", "_spallTrackID"];
-PARAMS_3(_gun,_type,_round);
-DEFAULT_PARAM(3,_doFragTrack,false);
-
 if (!GVAR(enabled)) exitWith {};
 
-//_enabled = getNumber (configFile >> "CfgAmmo" >> _type >> QGVAR(enabled));
-//if(_enabled < 1) exitWith {};
+private ["_enabled", "_doSpall", "_spallTrack", "_spallTrackID"];
+params ["_gun", "_type", "_round", ["_doFragTrack", false]];
+
 
 if(_round in GVAR(blackList)) exitWith {
     GVAR(blackList) = GVAR(blackList) - [_round];
 };
 
 // Exit on max track
-if( (count GVAR(objects)) > GVAR(MaxTrack)) exitWith { };
+if((count GVAR(objects)) > GVAR(MaxTrack)) exitWith {};
+
 
-if(_gun == ACE_player) then {
-    _doFragTrack = true;
-} else {
-    if((gunner _gun) == ACE_player) then {
-        _doFragTrack = true;
+if (_doFragTrack) then {
+    _doFragTrack = if (_gun == ACE_player) then {

```

Comment:
I strongly suggest
`if () then {_doFragTrack = } else {_doFragTrack = };`
over
`_doFragTrack = if () then {} else {};`

Scope should be no problem, because we private all variables.